VoodooGPIO/VoodooGPIOAMD.cpp

```diff
@@ -357,29 +357,27 @@ bool VoodooGPIOAMD::start(IOService *provider) {
 
     this->isInterruptBusy = true;
 
-    this->workLoop = getWorkLoop();
-    if (!this->workLoop) {
-        IOLog("%s::Failed to get workloop!\n", this->getName());
-        stop(provider);
+    workLoop = IOWorkLoop::workLoop();
+    if (!workLoop) {
+        IOLog("%s::Failed to set workloop!\n", getName());
         return false;
     }
     workLoop->retain();
 
     this->command_gate = IOCommandGate::commandGate(this);
     if (!this->command_gate || (this->workLoop->addEventSource(this->command_gate) != kIOReturnSuccess)) {
         IOLog("%s Could not open command gate\n", this->getName());
-        stop(provider);
```

Comment:
I assuming this change is because the kernel will call stop when `start` returns false?

Reply:
> Implementations of start must call start on their superclass at an appropriate point. If an implementation of start has already called super::start but subsequently determines that it will fail, it must call super::stop to balance the prior call to super::start and prevent reference leaks.
Guess not